server/services/incomingMovementsService.ts

```diff
@@ -9,9 +10,29 @@ export default class IncomingMovementsService {
     private readonly welcomeApiFactory: (token: string) => WelcomeApi
   ) {}
 
+  private moveTypeGroupingFunction(item: Movement): string {
+    if (item.moveType === 'PRISON_REMAND') return 'fromCourt'
+    if (item.moveType === 'PRISON_RECALL') return 'fromCustodySuite'
+    if (item.moveType === 'VIDEO_REMAND') return 'fromCustodySuite'
+    return null
+  }
+
+  private sortAlphabetically(movements: Movement[]): Movement[] {
+    return movements.sort((a, b) => {
+      const result = a.lastName.localeCompare(b.lastName)
+      return result !== 0 ? result : a.firstName.localeCompare(b.firstName)
+    })
+  }
+
   public async getIncomingMovements(agencyId: string): Promise<Movement[]> {
     const today = moment.now().toString()
     const token = await this.hmppsAuthClient.getSystemClientToken()
-    return this.welcomeApiFactory(token).getIncomingMovements(agencyId, today)
+    const movements = await this.welcomeApiFactory(token).getIncomingMovements(agencyId, today)
+    return this.sortAlphabetically(movements)
+  }
+
+  public async groupByMoveType(agencyId: string): Promise<Map<string, Movement[]>> {
+    const movements = await this.getIncomingMovements(agencyId)
+    return groupBy(movements, (movement: Movement) => this.moveTypeGroupingFunction(movement))
```

Comment:
very meh - 

if you look at the function `moveTypeGroupingFunction` , what it does and its signature, it doesn't need to be used as a grouping function - could be called something like `getMoveType`, `calculateMoveType`, `determineMoveType`... etc

Makes it a bit easier to reuse.

And super meh.. think you might be able to call this like: 
`groupBy(movements, this.moveTypeGroupingFunction)`
(possibly)

Reply:
grouping function renamed but wasn't able to call it as just `this.moveTypeGroupingFunction`